libunwind/src/CompartmentInfo.hpp

```diff
@@ -13,21 +13,68 @@
 #ifndef __COMPARTMENT_INFO_HPP__
 #define __COMPARTMENT_INFO_HPP__
 
+extern "C" {
+
+#include <link.h>
+
+// Must mirror the layout in rtld_c18n_machdep.h
+#if defined(__CHERI_PURE_CAPABILITY__) && defined(_LIBUNWIND_TARGET_AARCH64)
```

Comment:
```suggestion
#ifdef __CHERI_PURE_CAPABILITY__
#ifdef _LIBUNWIND_TARGET_AARCH64
```
ditto below, that way it's ready for the RISC-V version to be dropped in